From 399c77fc1a23f2d6cb56eaf4eb6469abe00412e6 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sun, 2 Oct 2022 12:36:16 +0200
Subject: [PATCH 45/52] Revert "elevator: add new field flags in struct
 elevator_queue"

This reverts commit aa75a302d13ecc6278c0227460f0cb0cf24b9cf3.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/elevator.c | 6 ++++--
 block/elevator.h | 4 +---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/block/elevator.c b/block/elevator.c
index 7cb61820c..c31976589 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -512,7 +512,7 @@ int elv_register_queue(struct request_queue *q, bool uevent)
 		if (uevent)
 			kobject_uevent(&e->kobj, KOBJ_ADD);
 
-		set_bit(ELEVATOR_FLAG_REGISTERED, &e->flags);
+		e->registered = 1;
 	}
 	return error;
 }
@@ -523,11 +523,13 @@ void elv_unregister_queue(struct request_queue *q)
 
 	lockdep_assert_held(&q->sysfs_lock);
 
-	if (e && test_and_clear_bit(ELEVATOR_FLAG_REGISTERED, &e->flags)) {
+	if (e && e->registered) {
 		struct elevator_queue *e = q->elevator;
 
 		kobject_uevent(&e->kobj, KOBJ_REMOVE);
 		kobject_del(&e->kobj);
+
+		e->registered = 0;
 	}
 }
 
diff --git a/block/elevator.h b/block/elevator.h
index 0b8240792..16cd8bded 100644
--- a/block/elevator.h
+++ b/block/elevator.h
@@ -100,12 +100,10 @@ struct elevator_queue
 	void *elevator_data;
 	struct kobject kobj;
 	struct mutex sysfs_lock;
-	unsigned long flags;
+	unsigned int registered:1;
 	DECLARE_HASHTABLE(hash, ELV_HASH_BITS);
 };
 
-#define ELEVATOR_FLAG_REGISTERED 0
-
 /*
  * block elevator interface
  */
-- 
2.38.0.rc1.6.g4fd6c5e444

